File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.cs
public class Solution {
    public int EvalRPN(string[] tokens) {
        Stack<int> stack = new Stack<int>();
        
        foreach (string token in tokens) {
            if (IsOperator(token)) {
                int rightOperand = stack.Pop();
                int leftOperand = stack.Pop();
                int result = PerformOperation(leftOperand, rightOperand, token);
                stack.Push(result);
            } else {
                stack.Push(int.Parse(token));
            }
        }
        
        return stack.Pop();
    }

    private bool IsOperator(string token) {
        return token == "+" || token == "-" || token == "*" || token == "/";
    }

    private int PerformOperation(int leftOperand, int rightOperand, string operatorToken) {
        switch (operatorToken) {
            case "+":
                return leftOperand + rightOperand;
            case "-":
                return leftOperand - rightOperand;
            case "*":
                return leftOperand * rightOperand;
            case "/":
                return leftOperand / rightOperand;
            default:
                throw new ArgumentException("Invalid operator");
        }
    }
}

File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.js
/**
 * @param {string[]} tokens
 * @return {number}
 */
var evalRPN = function(tokens) {
    let stack=[];
    for(let token of tokens){
        if(!isNaN(token)){stack.push(Number(token));}
        else{
            let b=stack.pop(),a=stack.pop();
            switch(token){
                case '+': stack.push(a+b);break;
                case '-': stack.push(a-b);break;
                case '*': stack.push(a*b);break;
                case '/': stack.push(Math.trunc(a/b));break;
            }
        }
    }return stack[0];
};